internal/provider/resources/resource_deployment.go

```diff
@@ -118,7 +118,7 @@ func (r *DeploymentResource) Create(
 		return
 	}
 
-	var diags diag.Diagnostics
+	diags := make(diag.Diagnostics, 0)
```

Comment:
For my own understanding - should these all be `[]diag.Diagnostics`? Or is the object itself an array?

Reply:
I guess the initialization value of 0 is just the "null" val here?

Reply:
`diag.Diagnostics` is equivalent to `[]diag.Diagnostic`

Reply:
so here its making an empty array